### PR TITLE
Fix root

### PR DESCRIPTION
**Pull request by telphan**

### CODE REVIEW

File: layers/meta-resin-jetson/recipes-kernel/linux/linux-tegra_%.bbappend [hunk 1]
```diff
@@ -14,24 +14,36 @@ RESIN_CONFIGS_remove = "brcmfmac"
...
+    rm -f ${D}/${KERNEL_IMAGEDEST}/extlinux/extlinux.conf_flasher
+    KERNEL_ROOTSPEC="${KERNEL_ROOTSPEC}" ; cat >${D}/${KERNEL_IMAGEDEST}/extlinux/extlinux.conf << EOF
 DEFAULT primary
 TIMEOUT 30
 MENU TITLE Boot Options
 LABEL primary
       MENU LABEL primary ${KERNEL_IMAGETYPE}
       LINUX /${KERNEL_IMAGETYPE}
-      APPEND ${KERNEL_ARGS} ${KERNEL_ROOTSPEC}
+      APPEND ${KERNEL_ARGS} $KERNEL_ROOTSPEC
```
Comment (floion) [inline layers/meta-resin-jetson/recipes-kernel/linux/linux-tegra_%.bbappend]: Is this intended?

Comment (telphan) [reply layers/meta-resin-jetson/recipes-kernel/linux/linux-tegra_%.bbappend]: Yes, so it takes the inline definition of the variable, not the yocto definition.

File: layers/meta-resin-jetson/recipes-kernel/linux/linux-tegra_%.bbappend [hunk 2]
```diff
@@ -14,24 +14,36 @@ RESIN_CONFIGS_remove = "brcmfmac"
...
+EOF
+    KERNEL_ROOTSPEC="${KERNEL_ROOTSPEC_FLASHER}" ; cat >${D}/${KERNEL_IMAGEDEST}/extlinux/extlinux.conf_flasher << EOF
+DEFAULT primary
+TIMEOUT 30
+MENU TITLE Boot Options
+LABEL primary
+      MENU LABEL primary ${KERNEL_IMAGETYPE}
+      LINUX /${KERNEL_IMAGETYPE}
+      APPEND ${KERNEL_ARGS} $KERNEL_ROOTSPEC
```
Comment (floion) [inline layers/meta-resin-jetson/recipes-kernel/linux/linux-tegra_%.bbappend]: Maybe use the same ${} convention as in the rest of the file when referencing variables?

Comment (telphan) [reply layers/meta-resin-jetson/recipes-kernel/linux/linux-tegra_%.bbappend]: Same case as above.

File: layers/meta-resin-jetson/recipes-kernel/linux/linux-tegra_%.bbappend (hunk 1 above)
Comment (floion) [reply layers/meta-resin-jetson/recipes-kernel/linux/linux-tegra_%.bbappend]: Both should expand the same variable. I'm not sure what you mean here

Comment (telphan) [reply layers/meta-resin-jetson/recipes-kernel/linux/linux-tegra_%.bbappend]: Not the same variable. KERNEL_ROOTSPEC="${KERNEL_ROOTSPEC}" I expand with $KERNEL_ROOTSPEC, and ${KERNEL_ROOTSPEC} expands the variable defined in the recipe.

**Pull request merged**